content/blog/2021-11-17-exp-versioning.md

```diff
@@ -0,0 +1,300 @@
+---
+title: Don't Just Track Your Experiments, Version Them
+date: 2021-11-17
+description: >
+  Versioning ML experiments combines the benefits of version control and
+  experiment tracking.
+descriptionLong: >
+  With experiment versioning, you can take ML experiment tracking to the next
+  level, making it easy and powerful to save, compare, and reproduce experiments
+  at scale in ways that neither traditional software version control nor
+  existing experiment tracking tools can.
+picture:
+pictureComment:
+author: dave_berenbaum
+commentsUrl:
+tags:
+  - MLOps
+  - DVC
+  - Experiments
+  - Experiment Tracking
+  - Experiment Versioning
+---
+
+Traditional software version control is not enough for tracking machine learning
+experiments. When tracking ML experiments, you need to not only version code,
+but to:
+
+- Log experiment information (like parameters and metrics).
+- Reproduce costly experiments (including data, code, models, and pipelines).
+- Organize and compare many experiments.
+
+Over the last few years, experiment tracking tools have come a long way so that
+users no longer need to log experiment info in spreadsheets or notebooks.
+Experiment tracking tools usually provide an API to log experiment information,
+a database to store it, and a dashboard to compare and visualize. We can learn
+from these tools and apply modern version control practices to:
+
+- Keep experiment information in your code repo so you can track changes
+  introduced by each experiment the same way you version code.
+- Codify your experiment pipeline (commands, inputs, and outputs) so you can
+  easily reproduce any experiment.
+- Associate experiments with commits in your repo so you can organize locally
+  and choose what to share.
+
+Experiment versioning builds on modern version control principles and technology
+to address experiment tracking needs.
+
+# Start Experiment Versioning
```

Comment:
I thought the same and wrote another version where I put this later. However, I struggled to keep it concrete and easy to understand since `exp init` does so much to introduce the project structure and pipeline. For example, how to mention codifying pipelines generically without showing anything DVC-specific? Again, ideas are welcome here!

Reply:
One good exercise might be to write the entire blog post on idea-level without DVC. Then, add the DVC as a last section. You have partially implemented this - but you can make one step further. Happy to brainstorm this together.